src/main/webapp/services/assignUserService.js

```diff
@@ -0,0 +1,104 @@
+import BaseService from './baseService';
+import { searchAssignableUser, assignUser } from '../api/apiCalls';
+import actionFactory from '../utils/actionFactory';
+
+const assignDialog = require('../templates/assignDialog.hbs');
+const errorDialog = require('../templates/errorDialog.hbs');
+const unexpectedErrorDialog = require('../templates/unexpectedErrorDialog.hbs');
+const forbiddenDialog = require('../templates/forbiddenDialog.hbs');
+const notFoundDialog = require('../templates/issueNotFoundDialog.hbs');
+const successDialog = require('../templates/successDialog.hbs');
+
+export default class AssignUserService extends BaseService {
+  constructor(serviceName) {
+    super(serviceName);
+    this.selectedUser = {};
+  }
+
+  successDialog(issueKey) {
+    this.closeDialog('assignIssue');
+
+    const template = successDialog({
+      issueKey,
+      prettyName: this.selectedUser.prettyName,
+    });
+
+    this.openDialog('userAssigned', template, {});
+  }
+
+  openAssignDialog(data, service) {
+    const template = assignDialog({
+      url: data.entity.issue.url,
+      key: data.entity.issue.key,
+      name: data.entity.issue.assignee.displayName,
+    });
+
+    const assignIssueAction = { type: 'assignIssue', label: 'ASSIGN' };
+    const closeDialogAction = { type: 'closeAssignDialog', label: 'Cancel' };
+
+    const userData = Object.assign({
+      user: {
+        service: service.serviceName,
+        crossPod: 'NONE',
+      },
+    }, actionFactory([assignIssueAction, closeDialogAction], service.serviceName, data.entity));
```

Comment:
Not very good for debugging

Reply:
Should I assign the result of this function to a variable?